sdk/CorrelationContext.php

```diff
@@ -59,7 +59,9 @@ private function removeCorrelationHelper(ContextKey $key, ?CorrelationContext $c
             return;
         }
 
-        $child->setParent($this->parent);
+        if ($child !== null && $this->parent !== null) {
```

Comment:
NPE on both the child and the parent.

Reply:
The complexity increase is because of this additional check, AFAIK?

Reply:
I find that to be an acceptable tradeoff.